src/components/panel/panel.js

```diff
@@ -355,6 +355,45 @@ angular
  * @param {!MdPanelPosition} position
  */
 
+/**
+ * @ngdoc method
+ * @name MdPanelRef#registerListener
+ * @description
+ *
+ * Registers a listener with the panel. The callback should return a promise,
+ * which will allow the action to continue when it gets resolved, or will
+ * prevent an action if it is rejected. Currently supported listener types:
+ * * `onClose` - Gets called before the panel begins closing.
+ *
+ * @param {string} type Type of listener.
```

Comment:
You should make an enum for the possible types to prevent user errors and allow better type checking. (Really useful in typescript typings.)